components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/InMemoryIdentityDataStore.java

```diff
@@ -145,6 +145,18 @@ public void remove(String userName, UserStoreManager userStoreManager) throws Id
         if (userName == null) {
             return;
         }
+        if (userStoreManager instanceof org.wso2.carbon.user.core.UserStoreManager) {
+            String caseInsensitiveUsername = ((org.wso2.carbon.user.core.UserStoreManager) userStoreManager)
+                    .getRealmConfiguration().getUserStoreProperty("CaseInsensitiveUsername");
+            if (!StringUtils.isBlank(caseInsensitiveUsername) && "true".equalsIgnoreCase(caseInsensitiveUsername.trim
+                    ())) {
```

Comment:
Instead of "true".equalsIgnoreCase(caseInsensitiveUsername.trim()), use Boolean.parseBoolean((caseInsensitiveUsername.trim())